pyproject.toml

```diff
@@ -91,12 +91,72 @@ exclude_lines = [
     "if __name__ == .__main__.:"
 ]
 
-[tool.isort]
-profile = "black"
-multi_line_output = 3
-include_trailing_comma = true
-force_grid_wrap = 0
-line_length = 120
+[tool.ruff]
+line-length = 120
+target-version = "py38"
+
+[tool.ruff.lint]
+select = [
+    "A", # flake8-builtins
+    #"ANN", # flake8-annotations
+    "ASYNC", # flake8-async
+    #"B", # flake8-bugbear
+    "BLE", # flake8-blind-except
+    #"COM", #flake8-commas
+    #"C4", #flake8-comprehensions
+    #"C90", # mccabe 
+    #"D", # pydocstyle
+    "DTZ", # flake8-datetimez
+    #"E", # pycodestyle (error)
+    "ERA", # eradicate
+    #"EM", # flake8-errmsg
+    "EXE", # flake8-executable
+    #"F", # Pyflakes
+    "FA", # flake8-future-annotations
+    #"FBT", # flake8-boolean-trap 
+    #"FIX", # flake8-fixme
+    "FLY", # flynt
+    "G", # flake8-logging-format
+    "I", # isort
+    #"ISC", # flake8-implicit-str-concat
+    "ICN", # flake8-import-conventions
+    "LOG", # flake8-logging
+    #"N", # pep8-naming
+    #"NPY", # NumPy-specific rules
+    "PD", # pandas-vet
+    "PERF", # Perflint 
+    "PIE", # flake8-pie
+    #"PT", # flake8-pytest-style
+    "PTH", # flake8-use-pathlib
+    #"FURB", # refurb
+    #"RET", # flake8-return
+    #"RUF", # Ruff-specific rules 
+    "S", # flake8-bandit
+    #"SIM", # flake8-simplify
+    "SLF", # flake8-self
+    "SLOT", # flake8-slots
+    #"TCH", # flake8-type-checking
+    #"TRY", # tryceratops 
+    "T20", # flake8-print
+    "T100", # flake8-debugger
+    #"UP", # pyupgrade
+    "W", # pycodestyle (warning)
+    "Q", # flake8-quotes
+    "YTT", # flake8-2020
+]
+isort.combine-as-imports = true
+pydocstyle.convention = "google"
+flake8-type-checking.exempt-modules = ["typing", "typing_extensions", "numpy", "networkx", "collections"]
+
+[tool.ruff.lint.extend-per-file-ignores]
+"test/**/*.py" = [
+    "S101", # asserts allowed in tests
+    "FBT", # Don't care about booleans as positional arguments in tests, e.g. via @pytest.mark.parametrize()
+    "D", # No doc requirements in tests
```

Comment:
`"W605" # Dont worry about invalid escape sequences (deprecated in Python 3.6)`

Reply:
Nice catch, added it to the ignore list.